spec.html

```diff
@@ -33100,15 +33101,17 @@ <h1>Date.prototype.getSeconds ( )</h1>
         <h1>Date.prototype.getTime ( )</h1>
         <p>This method performs the following steps when called:</p>
         <emu-alg>
-          1. Return ? thisTimeValue(*this* value).
+          1. Let _DateObject_ be ? ThisDateObject(*this* value).
+          1. Return _DateObject_.[[DateValue]].
         </emu-alg>
       </emu-clause>
 
       <emu-clause id="sec-date.prototype.gettimezoneoffset">
         <h1>Date.prototype.getTimezoneOffset ( )</h1>
         <p>This method performs the following steps when called:</p>
         <emu-alg>
-          1. Let _t_ be ? thisTimeValue(*this* value).
+          1. Let _DateObject_ be ? ThisDateObject(*this* value).
+          1. Let _t_ be _DateObject_.[[DateValue]].
```

Comment:
it's kind of weird to do this in two steps - why not a `ThisTimeValue` AO?

Reply:
Well, that's what we've already got (modulo the capitalization). And if we never needed access to `_DateObject_`, that would suffice. But since the `set` functions *do* need access to `_DateObject_`, we need `ThisDateObject` (or something even lower). And given that we have that, I'm not sure that encapsulating/hiding the `.[[DateValue]]` access is worth it.

Alternatively, we could abstract both the getting and setting, something like:
- `? GetTimeValue(*this* value)` and
- `SetTimeValue(*this* value, _t_)`

which would hide both `_DateObject_` and `[[DateValue]]` (mostly).

I leave it to the editors to make those choices.

Reply:
I'd expect that we'd use ThisTimeValue for the getting, and ThisDateObject for the setting.

Reply:
So are you suggesting that at each point we want to set the time value, we'd use:
```
1. Let _DateObject_ be ! ThisDateObject(*this* value).
1. Set _DateObject_.[[DateValue]] to ...
```
But wouldn't you then say that it's kind of weird to do *that* in two steps?

Reply:
Or did you mean: the "get" functions would use `ThisTimeValue` only, and the `set` functions would use `ThisDateObject` only?

Reply:
Since the get is way more common than the set, that’s the more important one to do in one step.

Reply:
@jmdyck Of those options, I prefer B, the way it is currently being done in this PR. C would be okay, but I don't like having to refer to the `*this*` value again.